plasmapy/particles/__init__.py

```diff
@@ -46,3 +50,10 @@
 deuteron = Particle("D 1+")
 triton = Particle("T 1+")
 alpha = Particle("He-4 2+")
+
+
+def from_json(json_string):
+    """
+    Returns the appropriate object from its JSON string representation
+    """
+    return json.loads(json_string, cls=ParticleJSONDecoder)
```

Comment:
I just realized this is a `json.loads`, while it should be `json.load`.  `json.load` will load from a json text file, while `json.loads` loads from a json string.  My understanding of this serialization functionality is to let a user save a particle object and reload it in 10 years.  In that case, `from_json` should definitely utilize `json.load` over `json.loads`.

Reply:
And then you can sort of get the `json.loads` functionality back through https://docs.python.org/3/library/io.html#text-i-o  !

Reply:
Ah right! That makes sense. However, how do we take care of multiple JSONs in the same file? This would cause an error since the `ParticleJSONDecoder` is only designed to process a single particle's JSON string at a time. We'd need some way of giving the `ParticleJSONDecoder` one string at a time. Any thoughts on this?

Reply:
Another thing that came to mind is that the user might want to store the file as they please. If I remember, @StanczakDominik had mentioned serializing particles that were in an xarray after a simulation. Using loads and dumps could potentially give users the benefit of flexibility to store the file as they please together with other simulation data in future. 
A possible function the user would write to store the particles would be something similar to:
```python
def put_my_particles_into_a_file(particle_list, filepath):
    particle_strings = list(map(lambda x: x.to_json(),particle_list))
    with open(filepath,"w") as particle_file:
         for particle in particle_strings:
              particle_file.write(particle)
```

A possible function the user would write to reload the particles would be something like this:
```python
def get_my_particles_back(filepath):
    particles_in_file = None
    particle_objects = []
    with open(filepath,"r") as particle_file:
         particles_in_file = particle_file.readlines()
    particle_objects = list(map(from_json,particles_in_file)) #converts each particle JSON string to respective particle object
    return particle_objects
```
A sample file would look like:
```txt
{"plasmapy_particle" : {..}}
{"plasmapy_particle" : {..}}
{"plasmapy_particle" : {..}}
```
Just a thought from my side. I'd love feedback on whether this is easy to use for the user.

Reply:
I think saving and loading multiple particles is another project in itself, and should be given it's own PR.  From my limited research, it looks like writing multiple python objects to a single json file is not a trivial task.  It looks like you have to bundle it all into one json object and write that.  Then we have to address the issue with unique json keys, since we're currently saving all particles with the `'plasmapy_particle'` key.  I do think we can leverage our `ParticleJSONDecoder` to pull out all the particle objects and return a list of particles; thus, no need to have anything else than `plasmapy.particles.from_json` to implement this functionality.

Reply:
I have a few ideas for this. I would love to share them and perhaps work on the one we agree upon to move forward with. I agree, a new PR is a good idea. 

Any further suggestions and advice are welcome! 

Thank you!

Reply:
Hey, I'm sorry I've been away from the project for a bit! I'm absolutely down to break accomplishing my xarray dreams into a few PRs!